### ISSUE TITLE
add documentation for running make on Windows

### ISSUE THREAD

**Issue by chrisjkuch** [drivendata/cookiecutter-data-science#333]

**Comment by mbkranz** [comment]
@chrisjkuch  Hi Chris! I'm sinking my teeth into the v2 on Windows. I'm wondering if you have any tips for this? I'm wondering if supporting docker containers would be the way to go/best practice or there are any other "best practices" or recommended ways of doing this? It seems like in the R world, there is some support for recommending docker ([see this R course here](https://reproducibility.rocks/materials/day4/01-docker))

**Comment by pjbull** [comment]
@mbkranz I was actually just working on some additional documentation for `make` on Windows.

Docker is one way to do it, but here are some others worth considering:

> This brings us to `make` compatibility on Windows systems, which has been a constant thorn in our sides. Our test suite actually runs on Windows with `make`. There are multiple ways to install `make` onto Windows systems. One that has received positive feedback is the version that comes with [Git for Windows](https://gitforwindows.org/). When you install git on Windows, the git bash shell comes with it, and handling your data science projects just in this environment is generally reliable. Alternatively, there are versions of `make` on the Windows package managers, for example [chocolately](https://community.chocolatey.org/packages/make), [winget](https://winget.run/pkg/GnuWin32/Make), and [scoop](https://scoop.sh/#/apps?q=make&id=c43ff861c0f1713336e5304d85334a29ffb86317).  Traditional ways to get the `make` binaries include [cygwin](https://www.cygwin.com/) and [MinGW](https://www.mingw-w64.org/). All these options work, but may come with some tradeoffs. Finally, [Windows Subsystem for Linux](https://learn.microsoft.com/en-us/windows/wsl/) is another great option. It gives a full, non-virtualized linux environment inside your windows machine. It can work well to run all your data science workflows through Ubuntu running on WSL.

Curious if any of those help and work well/easily for you.

**Comment by mbkranz** [comment]
Thanks for the great suggestions @pjbull . Let me know what you think about the below -- 
1. Git for windows
    - I use Git for Windows through [cmder emulator](https://cmder.app) and have used Git bash (and some of my colleagues use as well). 
    - Unfortunately, we aren't finding the `make` executable in the Git Bash shell... is there something we are missing by chance? I can't find the executable in any `Git` directory `bin` .... ![image](https://github.com/drivendata/cookiecutter-data-science/assets/14118123/cc78f5d5-13d4-4a74-bd91-511320c96e54)
2. WSL2
      - Definitely an option! I can definitely go this route for my personal use. However, my task is to try and develop recommendations/example analyses for a broad audienace of researchers/analysts. I'm worried WSL2 would be too much for folks unfamiliar with unix. Have you gotten feedback for using WSL2 with a more junior analyst type persona? 
3. Other options
      - Haven't tried these but thanks for the refs (make be good to just put a make binary executable in my cmder emulator folder. One consideration with these is that it may add an extra piece of software to download. I think git for windows is a MUST for all analysts (no matter what technical skill level). If `make` is included in `git for windows`, this would probably be the easiest way to run.

**Comment by pjbull** [comment]
Yeah, it does not seem like `make` actually is included with git bash anymore (if it ever was). It looks like there are prebuilt executables with `make` from the following places, though I can't vouch for them in any way:
 - [winlibs](https://winlibs.com/#download-release)
 - [ezwinports](https://sourceforge.net/projects/ezwinports/files/)

**Comment by jayqi** [comment]
I just saw that there's a package for Make on conda-forge, and it seems to include Windows builds. Has any tried this? https://anaconda.org/conda-forge/make/

**Comment by pjbull** [comment]
> I just saw that there's a package for Make on conda-forge, and it seems to include Windows builds. Has any tried this? https://anaconda.org/conda-forge/make/

I just tested this and it works, but with one slightly annoying caveat. Let's say I use conda to manage my virtual environments. Once I activate my project environment, I have to reinstall `make` into that environment as well since it is just in the base environment:

```
> conda install -c conda-forge make  # installs into base
> ccds

...  # setup project

> make create_environment
> conda activate project
> make requirements
'make' is not recognized as an internal or external command,
operable program or batch file.
```

**Comment by jayqi** [comment]
It'll probably take some thought here to figure out a good workflow to recommend. 

I think it's plausible that we might be able to recommend a workflow using a global application installer like [pixi](https://pixi.sh/latest/cli/#global-install) or [condax](https://github.com/mariusvniekerk/condax). We're in a bit of a transitionary period though where pixi is still building out functionality and popularity, and condax is a little unclear its long-term maintenance.

**Comment by jayqi** [comment]
Closed by #355
